osu.Game/Screens/Select/Carousel/CarouselBeatmapSet.cs

```diff
@@ -63,42 +63,74 @@ public override int CompareTo(FilterCriteria criteria, CarouselItem other)
             if (!(other is CarouselBeatmapSet otherSet))
                 return base.CompareTo(criteria, other);
 
+            int i = 0;
+
             switch (criteria.Sort)
             {
                 default:
                 case SortMode.Artist:
-                    return string.Compare(BeatmapSet.Metadata.Artist, otherSet.BeatmapSet.Metadata.Artist, StringComparison.OrdinalIgnoreCase);
+                    i = string.Compare(BeatmapSet.Metadata.Artist, otherSet.BeatmapSet.Metadata.Artist, StringComparison.OrdinalIgnoreCase);
+
+                    if (i != 0) return i;
+                    goto case SortMode.Title;
```

Comment:
Why does this need to cascade fallback like this, exactly? That makes little sense to me.

Surely you can use something like import date or something? Online ID even?

Reply:
Yeah, this is kinda overkill. I'll use Online ID instead.

Reply:
Online ID might not have been the best idea considering local maps. Creation date is probably best.

Reply:
Should I do this to all of them or just to `SortMode.Artist`, `SortMode.Title`, `SortMode.Author`, `SortMode.Source`, `SortMode.BPM`, `SortMode.Length` and `SortMode.Difficulty`?

Reply:
I'm not sure if rank date is unique enough that it can be ignored. The rest sounds fine to ignore.